eslint/codestyle.eslintrc.yml

```diff
@@ -1,26 +1,110 @@
 rules:
+  # ==== Naming Convention ====
+  # Enforces naming conventions for everything across a codebase
+  # TODO: configure
+  # '@typescript-eslint/naming-convention':
+  #  - warn
+  #  - selector: variable
+  #    format: [ camelCase, PascalCase, UPPER_CASE ]
+  #    leadingUnderscore: allow
+
   # ==== Spacing | Code style ====
 
-  # Enforce consistent brace style for blocks
+  # Warn about multiple empty lines
+  'no-multiple-empty-lines':
+    - warn
+    - max: 2
+      maxEOF: 1
+      maxBOF: 1
+
+  # Warn about spaces inside of parentheses
+  'space-in-parens':
+    - warn
+    - never
+
+  # Limit Cyclomatic Complexity
+  'complexity': warn
+
+  # Require following curly brace conventions
+  'curly':
+    - warn
+    - multi-line
+
+  # Max 3 classes per file (ideally 1 per file)
+  'max-classes-per-file':
+    - warn
+    - 3
+
+  # Warn about a maximum line length to be 160
+  'max-len':
+    - warn
+    - code: 160
+
+  # Require parentheses when invoking a constructor with no arguments
+  'new-parens': error
+
+  # Warn about empty block statements
+  'no-empty': warn
+
+  # Disallow mixed spaces and tabs for indentation
+  'no-mixed-spaces-and-tabs': error
+
+  # Warn about trailing whitespace at the end of lines
+  'no-trailing-spaces': warn
+
+  # Underscores allowed due to 'private-member' notation
+  'no-underscore-dangle': off
+
+  # `var` declaration operator is not allowed
```

Comment:
[cosmetic] we will use two kinds of quotes - back and single? because you are using singles on line 55

Reply:
Basically, I haven't found how to turn yml comments into JSDoc like, but 
 ` - is code quote in JSDoc
 ' / " - just quotes 
 
 But anyway updated, thanks!